src/NUnitTestAdapter/TestFilterBuilder.cs

```diff
@@ -0,0 +1,96 @@
+﻿// ***********************************************************************
+// Copyright (c) 2013-2015 Charlie Poole
+//
+// Permission is hereby granted, free of charge, to any person obtainingn
+// a copy of this software and associated documentation files (the
+// "Software"), to deal in the Software without restriction, including
+// without limitation the rights to use, copy, modify, merge, publish,
+// distribute, sublicense, and/or sell copies of the Software, and to
+// permit persons to whom the Software is furnished to do so, subject to
+// the following conditions:
+// 
+// The above copyright notice and this permission notice shall be
+// included in all copies or substantial portions of the Software.
+// 
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+// EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+// MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+// NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS BE
+// LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
+// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION
+// WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+// ***********************************************************************
+
+using System;
+using System.Collections.Generic;
+using System.Text;
+using NUnit.Engine;
+
+namespace NUnit.VisualStudio.TestAdapter
+{
+    public class TestFilterBuilder
+    {
+        public IList<string> Tests { get; set; }
+        public IList<string> Include { get; set; }
+        public IList<string> Exclude { get; set; }
```

Comment:
Yes, I just copied this and avoided changes for the moment. Currently, all filtering requests come to us from VS as a list of the test names - that is, VS does the filtering for us. We give categories and properties to VS as "traits" which it understands and it creates the test lists accordingly.

I agree that this could be simplified or maybe even eliminated as a class.